tests/unit/type/hash.tcl

```diff
@@ -647,6 +647,46 @@ start_server {tags {"hash"}} {
             assert {$len2 == $len3}
         }
     }
+    
+    test {HINCRBY over hash-max-listpack-value encoded with a listpack} {
+        r config set hash-max-listpack-value 8
+        
+        # bighash's value exceeds hash-max-listpack-value
+        r del smallhash bighash
+        r hset smallhash tmp 0
+        r hset bighash tmp 0
+        r hincrby smallhash tmp 10000000
+        r hincrby bighash tmp 100000000
```

Comment:
i only now realize that these are actually gonna get encoded in the listpack as integers (taking only 2 bytes), see `lpEncodeGetType`.
this means that even the existing check in HSET (`hashTypeTryConversion`) is wrong, so i guess we can let it slide, but let's at least drop the HINCRBY test, and keep only the one about HINCRBYFLOAT (in which we'll use actual fractional numbers, just for fun).

Reply:
we can use values that are floating point and don't have that effect (i.e. divisions of power-of-2).
e.g. `0.00000005`